lib/api/actions/functions.php

```diff
@@ -410,119 +411,184 @@ function beans_reset_action( $id ) {
+	// Bail out if the ID is not registered for the given status.
+	if ( false === _beans_get_action( $id, $status ) ) {
 		return false;
 	}
 
+	global $_beans_registered_actions;
 	unset( $_beans_registered_actions[ $status ][ $id ] );
 
 	return true;
-
 }
 
 /**
- * Merge action.
+ * Merge the action's configuration and then store it for the given ID and status.
  *
+ * If the action's configuration has not already be registered with Beans, just store it.
+ *
+ * @since  1.5.0
  * @ignore
+ * @access private
+ *
+ * @param string $id     A unique string used as a reference.
+ * @param array  $action The new action's configuration to merge and then store.
+ * @param string $status Status for which to merge/store this action.
+ *
+ * @return array
  */
-function _beans_merge_action( $id, $action, $status ) {
-
-	global $_beans_registered_actions;
-
-	$id = _beans_unique_action_id( $id );
+function _beans_merge_action( $id, array $action, $status ) {
+	$id                = _beans_unique_action_id( $id );
+	$registered_action = _beans_get_action( $id, $status );
 
-	if ( $_action = _beans_get_action( $id, $status ) ) {
-		$action = array_merge( $_action, $action );
+	// If the action's configuration is already registered with Beans, merge the new configuration with it.
+	if ( false !== $registered_action ) {
+		$action = array_merge( $registered_action, $action );
 	}
 
+	// Now store/register it.
 	return _beans_set_action( $id, $action, $status, true );
-
 }
 
 /**
- * Check all action status and return the current action.
+ * Get the current action, meaning get from the "added" and/or "modified" statuses.
  *
+ * @since  1.5.0
  * @ignore
+ * @access private
+ *
+ * @param string $id A unique string used as a reference.
+ *
+ * @return array|bool
  */
 function _beans_get_current_action( $id ) {
 
-	$action = array();
-
+	// Bail out if the action is "removed".
 	if ( _beans_get_action( $id, 'removed' ) ) {
 		return false;
 	}
 
-	if ( $added = _beans_get_action( $id, 'added' ) ) {
+	$action = array();
+
+	$added = _beans_get_action( $id, 'added' );
+	if ( false !== $added ) {
 		$action = $added;
 	}
 
-	if ( $modified = _beans_get_action( $id, 'modified' ) ) {
-		$action = array_merge( $action, $modified );
+	$modified = _beans_get_action( $id, 'modified' );
+	if ( false !== $modified ) {
+		$action = is_array( $action )
+			? array_merge( $action, $modified )
+			: $modified;
 	}
 
 	// Stop here if the action is invalid.
-	if ( 4 != count( $action ) ) {
+	if ( ! _beans_is_action_valid( $action ) ) {
 		return false;
 	}
 
 	return $action;
+}
 
+/**
+ * Validates the action's configuration to ensure "hook", "callback", "priority", and "args" are
+ * set and not null.
+ *
+ * @since  1.5.0
+ * @ignore
+ * @access private
+ *
+ * @param array $action Action's configuration.
+ *
+ * @return bool
+ */
+function _beans_is_action_valid( array $action ) {
+	return isset( $action['hook'], $action['callback'], $action['priority'], $action['args'] );
```

Comment:
This function replaces the `count( $action ) == 4` check.  It is now checking that each parameter (key) is set within the `$action`.